src/ethereum/EthereumNetwork.ts

```diff
@@ -1502,6 +1510,97 @@ export class EthereumNetwork {
     }
   }
 
+  /**
+   * Check the eth-balance-checker contract for balances
+   */
+  // @ts-expect-error
+  async checkEthBalChecker(): Promise<EthereumNetworkUpdate> {
+    const { builtinTokens, networkInfo, walletLocalData, currencyInfo } =
+      this.ethEngine
+    const { chainParams, rpcServers, ethBalCheckerContract } = networkInfo
+
+    const tokenBal: { [currencyCode: string]: string } = {}
+    // TODO: Consider just using a specific rpc server for ethers use
+    const jsonRpcUrl = rpcServers.find(
+      rpcServer =>
+        !rpcServer.includes('alchemyApiKey') &&
+        !rpcServer.includes('infuraProjectId') &&
+        !rpcServer.includes('poktPortalApiKey')
+    )
+    if (ethBalCheckerContract != null && jsonRpcUrl != null) {
+      const ethProvider = new ethers.providers.JsonRpcProvider(
+        jsonRpcUrl,
+        chainParams.chainId
+      )
+
+      const mainnetTokenAddr = '0x0000000000000000000000000000000000000000'
+      const tokenAddrs = [
+        ...Object.values(builtinTokens)
```

Comment:
Have to use allTokensMap instead of builtinTokens otherwise you won't lookup custom tokens